tests/EncryptedSessionsZF2/Session/SaveHandler/EncryptedSessionSaveHandlerTest.php

```diff
@@ -0,0 +1,202 @@
+
+        $this->mockSessionSaveHandler = $this->getMock(
+            'Zend\Session\SaveHandler\SaveHandlerInterface'
+        );
+
+        $this->mockEncryptionFilter = $this->getMock('Zend\Filter\Encrypt');
+        $this->mockDecryptionFilter = $this->getMock('Zend\Filter\Decrypt');
+
+        $this->encryptedSessionSaveHandler = new EncryptedSessionSaveHandler(
+            $this->mockSessionSaveHandler,
+            $this->mockEncryptionFilter,
+            $this->mockDecryptionFilter
+        );
+    }
+
+    /**
+     * @covers EncryptedSessionsZF2\Session\SaveHandler\EncryptedSessionSaveHandler::open
+     */
+    public function testEncryptedSessionSaveHandlerOpenIsCalled()
+    {
+        $this->mockSessionSaveHandler->expects($this->once())
+            ->method('open')
+            ->with($this->equalTo('testPath'), $this->equalTo('testName'))
+            ->will($this->returnValue(true));
+
+        $result = $this->encryptedSessionSaveHandler->open('testPath', 'testName');
+
+        $this->assertTrue($result);
+    }
+
+    /**
+     * @covers EncryptedSessionsZF2\Session\SaveHandler\EncryptedSessionSaveHandler::close
+     */
+    public function testEncryptedSessionSaveHandlerCloseIsCalled()
+    {
+        $this->mockSessionSaveHandler->expects($this->once())
+            ->method('close')
+            ->will($this->returnValue(true));
+
+        $result = $this->encryptedSessionSaveHandler->close();
+
+        $this->assertTrue($result);
+    }
+
+    /**
+     * @covers EncryptedSessionsZF2\Session\SaveHandler\EncryptedSessionSaveHandler::read
+     */
+    public function testEncryptedSessionSaveHandlerReadIsCalled()
+    {
+        $this->mockSessionSaveHandler->expects($this->once())
+            ->method('read')
+            ->with($this->equalTo('testId'))
+            ->will($this->returnValue('testData'));
+
+        $this->mockDecryptionFilter->expects($this->once())
+            ->method('filter')
+            ->with($this->equalTo('testData'))
+            ->will($this->returnValue('decryptedData'));
+
+        $result = $this->encryptedSessionSaveHandler->read('testId');
+
+        $this->assertEquals('decryptedData', $result);
+    }
+
+    /**
+     * @covers EncryptedSessionsZF2\Session\SaveHandler\EncryptedSessionSaveHandler::read
+     */
+    public function testEncryptedSessionSaveHandlerEmptyReadReturnsEmpty()
+    {
+        $this->mockSessionSaveHandler->expects($this->once())
+            ->method('read')
+            ->with($this->equalTo('testId'))
+            ->will($this->returnValue(''));
+
+        $this->mockDecryptionFilter->expects($this->never())
+            ->method('filter');
+
+        $result = $this->encryptedSessionSaveHandler->read('testId');
+
+        $this->assertEmpty($result);
+    }
+
+    /**
+     * @covers EncryptedSessionsZF2\Session\SaveHandler\EncryptedSessionSaveHandler::read
+     */
+    public function testEncryptedSessionSaveHandlerFalseReadReturnsFalse()
+    {
```

Comment:
Maybe this test could be phrased to indicate that it is ensuring that the decorator does not edit the value returned by the session handler.